lib/queue.js

```diff
@@ -26,16 +28,9 @@ module.exports = FetchQueue;
 
 FetchQueue.prototype = [];
 FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callback) {
```

Comment:
How do you feel about changing the signature of this function to accept just two arguments: `queueItem` and `callback`, @cgiffard? It would make it more intuitive to call and since we construct a queue item from the arguments anyway, it would let us shorten the method without having to add much code at all in the places where it's called